lib/constructs/provisioning-sfn-workflow.ts

```diff
@@ -0,0 +1,113 @@
+import * as sfn from "aws-cdk-lib/aws-stepfunctions";
+import * as lambdaNodeJs from "aws-cdk-lib/aws-lambda-nodejs";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { Construct } from "constructs";
+import { mapTasks, TasksMap } from "./sfn-lambda-invoke-task";
+
+/**
+ * Successful condition check used when status code is 200
+ */
+export const successResponse = sfn.Condition.numberEquals("$.cspResponse.Payload.statusCodeFn", 200);
+
+/**
+ * Client error condition check used when the status code is 400, 402, or 404
+ */
+export const clientErrorResponse = sfn.Condition.or(
+  sfn.Condition.numberEquals("$.cspResponse.statusCode", 400),
+  sfn.Condition.numberEquals("$.cspResponse.statusCode", 402),
+  sfn.Condition.numberEquals("$.cspResponse.statusCode", 404)
+);
+
+/**
+ * Internal error condition check used when the status code is 500
+ */
+export const internalErrorResponse = sfn.Condition.numberGreaterThanEquals("$.cspResponse.statusCode", 500);
+
+/**
+ * Retry condition check when the status code is 500, with a max retry of 6
+ */
+export const maxRetries = sfn.Condition.and(
+  sfn.Condition.numberGreaterThan("$.cspResponsePass.retryCount", 6),
+  sfn.Condition.numberGreaterThanEquals("$.cspResponse.statusCode", 500)
+);
```

Comment:
Do we want to avoid mention of the specific codes and values in the comments so that comments and implementation can't diverge in the future?

Reply:
Good point, avoided mentioning the status codes.